internal/filter/filter.go

```diff
@@ -100,6 +127,21 @@ func (f *Filter) ModuleReconcilerNodePredicate(kernelLabel string) predicate.Pre
 	)
 }
 
+func ModuleNMCReconcilerNodePredicate() predicate.Predicate {
+	return predicate.And(
+		skipDeletions,
```

Comment:
I think we'd want to delete a NMC if the corresponding node is gone.

Reply:
Currently i see it as an optimization (it does not affect the correctness of the code right now). So i would schedule it for the next PRs